hal/mbed_ticker_api.c

```diff
@@ -118,13 +118,22 @@ static void schedule_interrupt(const ticker_data_t *const ticker)
         // if the event at the head of the queue is in the past then schedule
         // it immediately.
         if (next_event_timestamp < present) {
-            relative_timeout = 0;
+            ticker->interface->fire_interrupt();
+            return;
         } else if ((next_event_timestamp - present) < MBED_TICKER_INTERRUPT_TIMESTAMP_MAX_DELTA) {
             relative_timeout = next_event_timestamp - present;
         }
     } 
 
-    ticker->interface->set_interrupt(ticker->queue->present_time + relative_timeout);
+    uint32_t new_match_time = ticker->queue->present_time + relative_timeout;
+    ticker->interface->set_interrupt(new_match_time);
+    // there could be a delay, reread the time, check if it was set in the past
+    // As result, if it is already in the past, we fire it immediately
+    update_present_time(ticker);
+    present = ticker->queue->present_time;
+    if (present > new_match_time) {
```

Comment:
If the ticker interrupt triggers when the present time increments from ```match_time -1``` -> ```match_time``` then this check (and the one above) needs to be ```if (present >= new_match_time)```.